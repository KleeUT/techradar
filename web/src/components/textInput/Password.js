import React from "react";
import TextInput from "./TextInput";

export default props => <TextInput type="password" {...props} />;
