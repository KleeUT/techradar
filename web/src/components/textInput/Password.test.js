import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Password from "./Password";
describe("Password Input: ", () => {
  it(" renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Password
        text={"With Text"}
        onChange={e => console.log()}
        label="password"
      />,
      div
    );
  });

  it(" passes snapshot tests", () => {
    const component = renderer.create(
      <Password
        text={"With some password"}
        onChange={e => console.log()}
        label="password"
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
