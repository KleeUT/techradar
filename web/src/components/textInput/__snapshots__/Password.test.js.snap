// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password Input:   passes snapshot tests 1`] = `
<div>
  <label
    className="sc-bwzfXH kEqaKq"
  >
    password
  </label>
  <input
    className="sc-bdVaJa etKSJQ"
    onChange={[Function]}
    placeholder="password"
    type="password"
    value="With some password"
  />
</div>
`;
